### DISCUSSION TITLE
Using NoneAsEmptyString causes missing values to trigger a DeserializeBodyError

### DISCUSSION THREAD

**Discussion by mu-arch** [jonasbb/serde_with#517] [Q&A] [answered]
I might just be missing something or there is some incompatability somewhere but when I use the 

#[serde_as(as = "NoneAsEmptyString")]

feature "" does become None as expected but now my true None values from a missing JSON property trigger a DeserializeBodyError. Suggestions appreciated.

My code: 
```rust
#[serde_as]
#[derive(Default, Debug, Clone, PartialEq, Serialize, Deserialize)]
pub struct MailParty {
    pub address: String,
    #[serde_as(as = "NoneAsEmptyString")]
    pub name: Option<String>,
}

**Comment by jonasbb** [comment, marked as answer]
It is really helpful in understanding the problem, if you can provide a runnable example including the serialized form and the expected outcome.

This is expected behavior of the serde derives. If you use `with` or `deserialize_with` on a field, the field becomes mandatory. The `as` and `deserialize_as` boil down to `with`/`deserialize_with`, so the same applies here. You need to add `#[serde(default)]` if you want to make the field optional.

**Reply by mu-arch** [reply]
Thank you very much that solved my problem and I learned something new ^-^ have a great day
